library/Vanilla/Formatting/Formats/HtmlFormat.php

```diff
@@ -0,0 +1,233 @@
+ * @copyright 2009-2019 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+namespace Vanilla\Formatting\Formats;
+
+use Garden\StaticCacheTranslationTrait;
+use Vanilla\Contracts\Formatting\FormatInterface;
+use Vanilla\Formatting\BaseFormat;
+use Vanilla\Formatting\Exception\FormattingException;
+use Vanilla\Formatting\Heading;
+use Vanilla\Formatting\Html\HtmlEnhancer;
+use Vanilla\Formatting\Html\HtmlPlainTextConverter;
+use Vanilla\Formatting\Html\HtmlSanitizer;
+use Vanilla\Formatting\Html\LegacySpoilerTrait;
+
+/**
+ * Format definition for HTML based formats.
+ */
+class HtmlFormat extends BaseFormat {
+
+    use StaticCacheTranslationTrait;
+
+    const FORMAT_KEY = "Html";
+
+    /** @var HtmlSanitizer */
+    private $htmlSanitizer;
+
+    /** @var HtmlEnhancer */
+    private $htmlEnhancer;
+
+    /** @var bool */
+    private $shouldCleanupLineBreaks;
+
+    /** @var HtmlPlainTextConverter */
+    private $plainTextConverter;
+
+    /**
+     * Constructor for dependency injection.
+     *
+     * @param HtmlSanitizer $htmlSanitizer
+     * @param HtmlEnhancer $htmlEnhancer
+     * @param HtmlPlainTextConverter $plainTextConverter
+     * @param bool $shouldCleanupLineBreaks
+     */
+    public function __construct(
+        HtmlSanitizer $htmlSanitizer,
+        HtmlEnhancer $htmlEnhancer,
+        HtmlPlainTextConverter $plainTextConverter,
+        bool $shouldCleanupLineBreaks = true
+    ) {
+        $this->htmlSanitizer = $htmlSanitizer;
+        $this->htmlEnhancer = $htmlEnhancer;
+        $this->plainTextConverter = $plainTextConverter;
+        $this->shouldCleanupLineBreaks = $shouldCleanupLineBreaks;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function renderHtml(string $content, bool $enhance = true): string {
+        $result = $this->htmlSanitizer->filter($content);
+
+        if ($this->shouldCleanupLineBreaks) {
+            $result = self::cleanupLineBreaks($result);
+        }
+
+        $result = $this->legacySpoilers($result);
+
+        if ($enhance) {
+            $result = $this->htmlEnhancer->enhance($result);
+        }
+        return $result;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function renderPlainText(string $content): string {
+        $html = $this->renderHtml($content, false);
+        return $this->plainTextConverter->convert($html);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function renderQuote(string $content): string {
+        $result = $this->htmlSanitizer->filter($content);
+
+        if ($this->shouldCleanupLineBreaks) {
+            $result = self::cleanupLineBreaks($result);
+        }
+
+        $result = $this->legacySpoilers($result);
+
+        // No Embeds
+        $result = $this->htmlEnhancer->enhance($result, true, false);
+        return $result;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function filter(string $content): string {
+        try {
+            $this->renderHtml($content);
```

Comment:
Nope. The intention is purely to ru the render and make sure nothing throws and exceptions. We don't actually save the rendered HTML on the forum, we save the raw body, so that is what gets passed through.

Reply:
Weird trick.